specs/registry-auth.md

```diff
@@ -0,0 +1,143 @@
+3. Notation requests the authorization service for a new bearer token for accessing the target resource with local credentials.
+4. The authorization grants and returns a new bearer token for access back to the notation client. The client-side cache is refreshed by the newly returned token.
+5. Notation attempts to access the remote registry again.
+6. The remote registry performs the requested operation and returns the response.
+
+#### Docker
+
+In the [Docker Token Authentication][token] specification, the *step 3* is implemented using a `GET` request where users are authenticated by the authorization service via `Basic` authentication scheme. Therefore, user credentials are only accepted in the form of username and password pair.
+
+#### OAuth 2
+
+Notation follows the [Docker Registry v2 authentication][oauth2] specification for the [OAuth 2.0][RFC6749] framework where the *step 3* is implemented using a `POST` request. Precisely, notation supports `password` and `refresh_token` grant types.
+
+**Note** Refresh tokens are often known as identity tokens in the context of registry authentication.
+
+
+## Credential Store
+
+As local credentials may be required to access the remote registries, they need to be stored and accessed securely. To achieve maximum security, credential helpers are preferred so that credentials are stored in the system key chain with better protection. If credential helpers are not available, notation will fall back to credential files with proper access control.
+
+### Credential Helper
+
+To achieve maximum compatibility with existing systems, [docker credential helpers](https://github.com/docker/docker-credential-helpers) and its [protocol](https://docs.docker.com/engine/reference/commandline/login/#credential-helper-protocol) are adopted as the credential helpers for `notation`.
+
+The credential store can be specified globally or per registry by setting the notation config.
+
+```json
+{
+    "credHelpers": {
+        "registry.wabbit-networks.io": "wabbithelper",
+        "another.wabbit-networks.io": "foobar"
+    },
+    "credsFile": "/absolute/path/to/auth.json",
+    "credsStore": "whatever"
+}
+```
+
+**Note** The absolute path to credential file `credsFile` is used to store extra metadata by credential helper drivers such as [docker/cli](https://github.com/docker/cli/blob/master/cli/config/credentials/native_store.go).
+
+### Credential File
+
+The credential file is alternative credential store when credential helpers are not available. The default file path is
+
+```
+{CONFIG}/notation/auth.json
+```
+
+The credential file path can be altered by setting the `credsFile` field of the notation config.
+
+```json
+{
+    "credsFile": "/absolute/path/to/auth.json"
+}
+```
+
+Since credentials are stored in plaintext, the permission of the credential file MUST be kept minimum when storing credentials. On Unix / Linux, the permission MUST be either `0600` (default) or `0400` (read-only).

```

Comment:
Do we need to support local credential file, given cred helper provide a easy path to integrate both local and remote credentials?

Reply:
Tracked in #206